dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/about/action/RedirectAction.java

```diff
@@ -77,13 +77,21 @@ public String execute()
                     if ( app.getShortName().equals( startModule.substring( "app:".length() ) ) )
                     {
                         redirectUrl = app.getLaunchUrl();
+                        if ( redirectUrl.endsWith( "dhis-web-dataentry/" ) )
+                        {
+                            redirectUrl += "index.action";
+                        }
```

Comment:
Just curious, if the `startModule` is a Struts module/the data entry module, does it start with `app:`?

Just wondering what the if / else logic is good for. I.e. if the data entry module starts with `app:`, then we don't need to put this special fix in both branches of the if / else.

```
if ( startModule.startsWith( "app:" ) )
```

Reply:
I think it does not but I could not debug the class as I don*t have a working IDE config for the web modules.

Reply:
Okay, perhaps just a good old system.out.println?

Reply:
removed it from that branch and tested it still works which it does.